Dockerfile

```diff
@@ -1,21 +1,54 @@
-FROM ubuntu:bionic
+# syntax=docker/dockerfile:experimental
 
-# Set up environment
-ENV LANG C.UTF-8
+# Build stage: Install python dependencies
+# ===
+FROM ubuntu:focal AS python-dependencies
+RUN apt-get update && apt-get install --no-install-recommends --yes python3-pip python3-setuptools
+ADD requirements.txt /tmp/requirements.txt
+RUN --mount=type=cache,target=/root/.cache/pip pip3 install --user --requirement /tmp/requirements.txt
+
+
+# Build stage: Install yarn dependencies
+# ===
+FROM node:12-slim AS yarn-dependencies
 WORKDIR /srv
+ADD package.json .
+RUN --mount=type=cache,target=/usr/local/share/.cache/yarn yarn install
 
-# System dependencies
-RUN apt-get update && apt-get install -y --no-install-recommends python3 python3-setuptools python3-pip
+# Build stage: Run "yarn run build-js"
+# ===
+FROM yarn-dependencies AS build-js
+ADD static/js static/js
+ADD webpack.config.js .
+RUN yarn run build-js
 
-# Set git commit ID
-ARG COMMIT_ID
-ENV COMMIT_ID "${COMMIT_ID}"
-ENV TALISKER_REVISION_ID "${COMMIT_ID}"
+# Build stage: Run "yarn run build-css"
+# ===
+FROM yarn-dependencies AS build-css
+ADD static/sass static/sass
+RUN yarn run build-css
+
+# Build the production image
+# ===
+FROM ubuntu:focal
 
-# Import code, install code dependencies
 COPY . .
-RUN python3 -m pip install --no-cache-dir -r requirements.txt
+# Install python and import python dependencies
+RUN apt-get update && apt-get install --no-install-recommends --yes python3 python3-lib2to3 python3-pkg-resources ca-certificates libsodium-dev
+COPY --from=python-dependencies /root/.local/lib/python3.8/site-packages /tmp/site-packages
```

Comment:
`COPY --from=python-dependencies /root/.local/lib/python3.8/site-packages /root/.local/lib/python3.8/site-packages`

This looks like stuff from the previous state of the dockerfile.

Reply:
What do you mean? Did we not agree to hardcode the python version?

Reply:
We did but the line 38 looks wrong, you are copying to `/tmp`

Reply:
ohh ok I see what you mean